### PR TITLE
Document FIPS image and Openshift default namespace limitation

### LINKED ISSUE #6332: Add public FIPS documentation
## Proposal

We recently merged [FIPS changes](https://github.com/elastic/cloud-on-k8s/pull/6071) in which we also release a `FIPS` enabled version of the ECK Operator during the standard release process.  We need to add some public documentation for this feature.

### PR DESCRIPTION
**Pull request by naemono**
closes #6332 

This change adds public documentation for the FIPS image that was released with the 2.6 version of the ECK Operator. This also adds additional documentation surrounding an Openshift limitation on using the default namespace.

### CODE REVIEW

File: docs/advanced-topics/openshift.asciidoc [hunk 1]
```diff
@@ -52,6 +52,8 @@ oc adm pod-network make-projects-global elastic-system
 
 . Create a namespace to hold the Elastic resources ({eck_resources_list}):
 +
+NOTE: A namespace outside of the `default` namespace is required such that default https://docs.openshift.com/container-platform/4.1/authentication/managing-security-context-constraints.html[Security Context Constraint] (SCC) permissions are applied automatically. Elastic resources will not work in the `default` namespace, using the `default` service account.
```
Comment (pebrc) [inline docs/advanced-topics/openshift.asciidoc:55]:
```suggestion
NOTE: A namespace other than the `default` namespace is required such that default https://docs.openshift.com/container-platform/4.1/authentication/managing-security-context-constraints.html[Security Context Constraint] (SCC) permissions are applied automatically. Elastic resources will not work in the `default` namespace, using the `default` service account.
```

This is fine but the OpenShift docs are much more prescriptive: 

> You cannot assign a SCC to pods created in one of the default namespaces: default, kube-system, kube-public, openshift-node, openshift-infra, openshift. These namespaces should not be used for running pods or services.

File: docs/advanced-topics/fips.asciidoc [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+[source,sh]
+----
+helm install elastic-operator elastic/eck-operator \
+  -n elastic-system --create-namespace \
+  --set=image.repository=docker.elastic.co/eck/eck-operator-fips
+----
+
+=== FIPS compliant installation using manifests
+
+The `StatefulSet` will need to be patched after installation using manifests to append `-fips` to the `spec.template.spec.containers[*].image` to install a FIPS compliant version of the ECK Operator.  Reference <<{p}-deploy-eck,Instsall ECK using manifests>> for full manifest installation instructions.
```
Comment (pebrc) [inline docs/advanced-topics/fips.asciidoc:32]: Installing a non-FIPS version first to later patch it might not be acceptable for some customers. Maybe rephrase by giving instructions how to replace the image before installation e.g. with a simple text editor or a `sed` command or similar. We can still leave the path option in the docs for users wanting to transition to FIPS.

File: docs/advanced-topics/fips.asciidoc [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+The Federal Information Processing Standard (FIPS) Publication 140-2, (FIPS PUB 140-2), titled "Security Requirements for Cryptographic Modules" is a U.S. government computer security standard used to approve cryptographic modules. Since version 2.6 ECK offers a FIPS compliant image that is a drop-in replacement for the standard image.
+
+For the ECK operator, adherence to FIPS 140-2 is ensured by
+
+- Using FIPS approved / NIST recommended cryptographic algorithms
+- Compiling the operator using the link:https://github.com/golang/go/blob/dev.boringcrypto/README.boringcrypto.md[BoringCrypto] library for various cryptographic primitives.
+
+== Installation
+
+=== FIPS compliant installation using Helm
```
Comment (Kushmaro) [inline docs/advanced-topics/fips.asciidoc:19]: might not want to use "FIPS compliant" wording - I'd prefer, FIPS-enabled
reason is, we can't really ensure "compliance" we can ensure that the operator uses the right libs and encryption in the (very) specific context of the operator, but customers requiring FIPS compliance on their ECK install (not just operator)  need to do much more than just use the FIPS-enabled operator.

so this is really just to avoid customer confusion.

File: docs/advanced-topics/fips.asciidoc [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+=== FIPS compliant installation using manifests
+
+The `StatefulSet` definition within the yaml installation manifest will need to be patched prior to installation to append `-fips` to the `spec.template.spec.containers[*].image` to install a FIPS-enabled version of the ECK Operator.  Reference <<{p}-deploy-eck,Instsall ECK using manifests>> for full manifest installation instructions.
+
+NOTE: `${ECK_VERSION}` in the following command needs to be replaced with the version of the Operator that is to be installed.
+
+[source,sh]
+----
+curl -s https://download.elastic.co/downloads/eck/${ECK_VERSION}/operator.yaml | sed 's/\(.*image:\).*$/\1 "docker.elastic.co\/eck\/eck-operator-fips:${ECK_VERSION}"/g' | kubectl apply -f -
```
Comment (thbkrkr) [inline docs/advanced-topics/fips.asciidoc:38]: Another sed command proposal: a bit shorter, `-r` to avoid to escape `(` and `#`as separator to avoid to write `/\`. 

```suggestion
curl -s https://download.elastic.co/downloads/eck/${ECK_VERSION}/operator.yaml | sed -r 's#(image:.*eck-operator.*)"#\1-fips"#' | kubectl apply -f -
```

Comment (naemono) [reply docs/advanced-topics/fips.asciidoc]: I'm great with this.  Thanks @thbkrkr !

File: docs/advanced-topics/openshift.asciidoc [hunk 5]
```diff
@@ -52,6 +52,8 @@ oc adm pod-network make-projects-global elastic-system
 
 . Create a namespace to hold the Elastic resources ({eck_resources_list}):
 +
+NOTE: A namespace other than the default namespaces: default, kube-system, kube-public, openshift-node, openshift-infra, openshift is required such that default https://docs.openshift.com/container-platform/4.1/authentication/managing-security-context-constraints.html[Security Context Constraint] (SCC) permissions are applied automatically. Elastic resources will not work properly in any of the default namespaces.
```
Comment (barkbay) [inline docs/advanced-topics/openshift.asciidoc:55]: Namespaces (a.k.a `projects` in OpenShift) provided by OpenShift starts with `openshift-`. Maybe we should reflect that naming? (see https://docs.openshift.com/online/pro/architecture/core_concepts/projects_and_users.html#architecture-projects-projects-at-install)

Comment (thbkrkr) [reply docs/advanced-topics/openshift.asciidoc:55]: I'm good with the current version but I find Barkbay's comment interesting, hence the following proposition:
```suggestion
NOTE: A namespace other than the default namespaces (default, kube-*, openshift-*, etc) is required such that default https://docs.openshift.com/container-platform/4.1/authentication/managing-security-context-constraints.html[Security Context Constraint] (SCC) permissions are applied automatically. Elastic resources will not work properly in any of the default namespaces.
```

Comment (thbkrkr) [reply docs/advanced-topics/openshift.asciidoc:55]: Also, I spotted that the URL uses an old version: https://docs.openshift.com/container-platform/4.1/. We could use the latest https://docs.openshift.com/container-platform/4.12.
This is unrelated to this PR but there is another openshift URL that we could update in `docs/advanced-topics/custom-images.asciidoc`.

Comment (naemono) [reply docs/advanced-topics/openshift.asciidoc:55]: Yeah I was going to get @barkbay comments addressed here shortly. I'll check your proposition. Thanks @thbkrkr

Comment (naemono) [reply docs/advanced-topics/openshift.asciidoc]: @thbkrkr I've updated the wording, using your recommendation, and updating all of the openshift links to v4.12.

**Review by thbkrkr — APPROVED** [review]
Left one last nitpick, otherwise LGTM.
